server/datastore/mysql/apple_mdm.go

```diff
@@ -1674,6 +1674,71 @@ func (ds *Datastore) GetMDMAppleBootstrapPackageBytes(ctx context.Context, token
 	return &bp, nil
 }
 
+// TODO(Sarah): Use constants to map ncr.status to bootstrap package status according to Apple
+// specs. https://developer.apple.com/documentation/devicemanagement/installenterpriseapplicationresponse
+func (ds *Datastore) GetMDMAppleBootstrapPackageSummary(ctx context.Context, teamID uint) (*fleet.MDMAppleBootstrapPackageSummary, error) {
+	stmt := `
+          SELECT
+              COUNT(IF(ncr.status = 'Acknowledged', 1, NULL)) AS installed,
+              COUNT(IF(ncr.status = 'Error', 1, NULL)) AS failed,
+              COUNT(IF(ncr.status IS NULL OR (ncr.status != 'Acknowledged' AND ncr.status != 'Error'), 1, NULL)) AS pending
+          FROM
+              hosts h
+          LEFT JOIN host_mdm_apple_bootstrap_packages hmabp ON
+              hmabp.host_uuid = h.uuid
+          LEFT JOIN nano_command_results ncr ON
+              ncr.command_uuid  = hmabp.command_uuid
+          JOIN host_mdm hm ON
+              hm.host_id = h.id
+          WHERE
+              hm.installed_from_dep = 1 AND COALESCE(h.team_id, 0) = ?`
+
+	var bp fleet.MDMAppleBootstrapPackageSummary
+	if err := sqlx.GetContext(ctx, ds.reader, &bp, stmt, teamID); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get bootstrap package summary")
+	}
+	return &bp, nil
+}
+
+func (ds *Datastore) RecordHostBootstrapPackage(ctx context.Context, commandUUID string, hostUUID string) error {
+	stmt := "INSERT INTO host_mdm_apple_bootstrap_packages (command_uuid, host_uuid) VALUES (?, ?)"
+	_, err := ds.writer.ExecContext(ctx, stmt, commandUUID, hostUUID)
+	return ctxerr.Wrap(ctx, err, "record bootstrap package command")
+}
+
+// TODO(Sarah): Use constants to map ncr.status to bootstrap package status according to Apple
+// specs. https://developer.apple.com/documentation/devicemanagement/installenterpriseapplicationresponse
+func (ds *Datastore) GetHostMDMMacOSSetup(ctx context.Context, hostID uint) (*fleet.HostMDMMacOSSetup, error) {
+	// TODO(Sarah): Is ncr.result the correct column to use here? I don't see where error details are stored.
```

Comment:
I don't think error details are stored in nanomdm tables. If you look in the `nanomdm` repo, you'll see that the `ErrorChain` field in the mdm command struct is not used anywhere. We do explicitly save it in the fleet repo for install/remove profile commands (in the `host_mdm_apple_profiles` table), but not for other commands AFAIK.

Reply:
> status is inferred from the result, and stored as a top-level column for convenience.

@roperzh, isn't `ncr.status` taken directly from the device [response](https://developer.apple.com/documentation/devicemanagement/installenterpriseapplicationresponse) rather than being inferred from `result`?

Reply:
> Having that said, @gillespi314 do you mind clarifying why you're considering using ncr.result instead of ncr.status

We need to include the error in the API response detail.

Reply:
It looks like this might be related to a [bug](https://github.com/fleetdm/fleet/issues/11040) with command results generally.

Reply:
> @roperzh, isn't ncr.status taken directly from the device [response](https://developer.apple.com/documentation/devicemanagement/installenterpriseapplicationresponse) rather than being inferred from result?

IIRC they are the same thing, that response is what get stored in `result`.

> We need to include the error in the API response detail.

gotcha, thanks. Does it make sense to use  `status` to filter and also query `result`? then you can unmarshal the XML and get the error, at least for now.

> It looks like this might be related to a https://github.com/fleetdm/fleet/issues/11040 with command results generally.

I think that's a separate issue, the issue you linked seems to be that we're validating the XML payload (good) but the error message is not helpful